### PR TITLE
Read data in partition, output to stdout or file

### PR DESCRIPTION
**Pull request by PizieDust**
part of #14 

## Outline
- This PR introduces a new executable `read_partition.ml` which reads content stored in a partition. 
- This executable uses the `cmdliner` package to parse and read arguments passed when calling this executable and also generates a lovely manual which can be called by adding the `--help` flag (I love this package)
- Basically we read in a disk image, unmarshal it's MBR header to get the partition table, grab the specific partition we want to read from. After getting this partition, we calculate information about it's sectors and then read the data from these sectors.

## Usage
After building, the command can be called with:
```sh
read_partition.exe test.img 2
```
- `test.img` : The disk image which contains the partition.
- `2` : The partition number we want to read from. 
Running this command will print the contents of partition 2 to stdout.

To output the contents to a file:
```sh
read_partition.exe -f file.txt test.img 2
```
This will output the contents to a file called `file.txt`. If this file doesn't exist, the command will create the file. 

## Simple Test
- Create a new text file with some text:
 ```sh
echo "Simple file for testing" > sample.txt
```
- Create a new disk image with 3 partitions:
```sh
fallocate -l test.img 2M`
parted --align none test.img mklabel msdos
parted --align none test.img mkpart primary 1s 2s
parted --align none test.img mkpart primary 3s 4s
parted --align none test.img mkpart primary 5s 6s
```

- Add the text file to the partition
```sh
dd if=sample.txt of=test.img bs=512 seek=1 conv=notrunc
```
- Run the command now to read file from the partition:
```sh
read_partition.exe test.img 1`
```

cc @reynir

### CODE REVIEW

File: bin/read_partition.ml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  let () = really_input ic buffer 0 (num_sectors * sector_size) in
+  close_in ic;
+  buffer
+
+let extract_partition_data mbr partition_num output_file =
+  let partition = get_partition_info mbr partition_num in
+  let start_sector, num_sectors, sector_size =
+    calculate_partition_info partition
+  in
+  let buffer = read_partition_data mbr start_sector num_sectors sector_size in
```
Comment (reynir) [inline bin/read_partition.ml:42]: We should incrementally read and write the data as the partition could potentially be larger than what we can fit in memory.

**Review by reynir — COMMENTED** [review]
Nice!

```
$ fallocate -l test.img 2M
$ parted --align none test.img mklabel msdos
$ parted --align none test.img mkpart primary 1s 20s
$ tar -cf tmp.tar LICENSE
$ dd if=test.tar of=test.img seek=1 conv=notrunc
20+0 records in
20+0 records out
10240 bytes (10 kB, 10 KiB) copied, 0.000348219 s, 29.4 MB/s
$ dune exec -- bin/read_partition.exe test.img  1 | tar -t
LICENSE
```

File: bin/read_partition.ml (hunk 1 above)
Comment (PizieDust) [reply bin/read_partition.ml]: I've updated it to use a chunk size of 1024 bytes. let me know if to use a bigger chunk size.

**Comment by reynir** [comment]
Thanks!

**Pull request merged**